File: articles/data-measurement-reproducibility-life-sciences/python/uncertainty_budget.py
"""
Measurement uncertainty budget workflow.

Run:
    python python/uncertainty_budget.py
"""

from pathlib import Path

import pandas as pd

from reproducibility_core import uncertainty_budget


ARTICLE_DIR = Path(__file__).resolve().parents[1]
DATA_PATH = ARTICLE_DIR / "data" / "uncertainty_components.csv"


def main() -> None:
    components = pd.read_csv(DATA_PATH)
    summary = uncertainty_budget(components, coverage_factor=2.0)

    print(components.to_string(index=False))
    print(summary.round(5).to_string(index=False))


if __name__ == "__main__":
    main()

File: articles/data-measurement-reproducibility-life-sciences/r/uncertainty_budget.R
# Measurement uncertainty budget workflow.

data_path <- file.path("data", "uncertainty_components.csv")
if (!file.exists(data_path)) {
  data_path <- file.path("..", "data", "uncertainty_components.csv")
}

components <- read.csv(data_path)

combined_standard_uncertainty <- sqrt(sum(components$standard_uncertainty^2))
coverage_factor <- 2.0
expanded_uncertainty <- coverage_factor * combined_standard_uncertainty

summary_df <- data.frame(
  combined_standard_uncertainty = combined_standard_uncertainty,
  coverage_factor = coverage_factor,
  expanded_uncertainty = expanded_uncertainty
)

print(components)
print(round(summary_df, 5))
